src/AdsClient/Internal/SocketExtensions.cs

```diff
@@ -10,18 +10,18 @@ namespace Viscon.Communication.Ads.Internal;
 /// </remarks>
 internal static class SocketExtensions
 {
-    public static SocketAwaitable ReceiveAsync(this Socket socket, SocketAwaitable awaitable)
+    public static SocketAwaitable ReceiveAwaitable(this Socket socket, SocketAwaitable awaitable)
```

Comment:
Needed to change the name, otherwise the compiler won't pick that extension method, as an overload on `Socket` itself would be choosen.